src/core/PixelStream.cpp

```diff
@@ -191,20 +207,34 @@ void PixelStream::adjustSegmentRendererCount(const size_t count)
     {
         segmentRenderers_.clear();
         for (size_t i=0; i<count; ++i)
-            segmentRenderers_.push_back( PixelStreamSegmentRendererPtr(new PixelStreamSegmentRenderer(renderContext_)) );
+            segmentRenderers_.push_back( boost::make_shared<PixelStreamSegmentRenderer>( ));
     }
 }
 
+void PixelStream::refreshSegmentsList( const deflect::PixelStreamSegments& segments )
+{
+    qDeleteAll( segmentsList_ );
```

Comment:
that does operator delete? This happens every frame? new/delete every frame should be avoided if possible.